docs/database/database-spans.md

```diff
@@ -268,6 +287,84 @@ Placeholders in a parameterized query SHOULD not be sanitized. E.g. `where id =
 e.g. from `IN (?, ?, ?, ?)` to `IN (?)`, as this can help with extremely long IN-clauses,
 and can help control cardinality for users who choose to (optionally) add `db.query.text` to their metric attributes.
 
+## Synthetic query text
+
+The `db.query.synthetic` attribute captures a shortened representation of a query text
+which SHOULD have low-cardinality and SHOULD NOT contain any dynamic or sensitive data.
+
+It SHOULD only be reported if instrumentation parses query text for the database call.
+
+When query text is parsed, the parser SHOULD extract a list of operations and corresponding
+collection names performed in this query.
+
+The `db.query.synthetic` attribute value SHOULD contain a semicolon separated list
+of operation and corresponding collection names formatted in the following way:
+
+```
+{operation1} {collection1}; {operation2} {collection2}; ...
+````
```

Comment:
wdyt of defining `db.query.synthetic` as a super sanitized version of `db.query.text` where the only things that are kept are operation names and collection names?

so that

> SELECT * from TABLE1, TABLE2 where ...

would become

> SELECT TABLE1 TABLE2

Reply:
Using empty space as a separator may cause problems, as databases may allow table names to have empty spaces.

For instance, in MySQL, this statement is allowed:

```sql
CREATE TABLE `USER TABLE` (
    UserID INT
);
```

For a query like this

```sql
SELECT * from `USER TABLE`, A;
```

would become

```sql
SELECT USER TABLE A
```

Which seems confusing, as we might think it queries three tables.

---

If we use a comma as a separator, we got 

```sql
SELECT USER TABLE, A
```

This seems clear.

Reply:
It's a good point! Though I don't think anyone could make much sense out of this attribute alone or parse it in any meaningful way.

It's just a grouping key that is also sometimes readable for human beings.  I expect that the main use case for it is to get top N longest queries and then for each of them look into `db.query.text` to see what was this query about (via span attributes or metric exemplars). 

So I'm not sure if we need to worry about spaces in the table names on this attribute.

Reply:
> Though I don't think anyone could make much sense out of this attribute alone or parse it in any meaningful way.

It sounds like it is a hashed-alike string with a human-readable context. There may be a use case to see the duration among different `db.query.synthetic`, so we can quickly find the slowest `db.query.synthetic` (a pattern) from a table perspective instead of a single query perspective.

This would require us to determine the exact database tables from `db.query.synthetic`. With `SELECT USER TABLE A`. We might think there are three tables: `USER`, `TABLE, ' and `A`. Or, two tables: `USER`, `TABLE A`, etc.

I know using empty spaces should be fine in most cases, but I think we should keep our design more tolerant of rare cases. Just like what we did for `db.collection.name`.

Reply:
Another option: Users can add `"` to the database table for this special case. So, it can look like

```sql
SELECT "USER TABLE" "A"
```

wdyt? Would it sound too complicated?

Reply:
would it work if we only quote tables that have spaces?

Reply:
what would we expect the `db.collection.name` be in the case of `` `USER TABLE` `` ? I'd expect it to contain all the characters that were provided in the query including quotes, double-quotes, backticks, non-ascii chars. etc.

That aligns with the language we have for collection name:
> to capture the value as provided by the application without attempting to do any case normalization

Then it should be represented in the same way in the `db.query.summary` - I can add some wording that targets there should follow the same as-is principle as on the attributes